### DISCUSSION TITLE
How to filter based on array of values

### DISCUSSION THREAD

**Discussion by lbensaad** [pocketbase/pocketbase#3156] [Q&A] [answered]
I have a collection user that has a multi-relation to itself named friends. If I have a user `{id="7", friends=["1","2","5"]}` how to filter user collection to get the friends of user 7 for example?
I tried `filter: 'id in ["1","2","5"]'` and `filter: 'id ?= ["1","2","5"]'` but it did not work.
Please tell me what is the exact filter value for this example other then `filter: 'id = "1" || id="2" || id="3"'`?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no support for direct `IN` operator at the moment and you have to use the `||` combined expression. If the ids in your example is a JS array you can combine it with `map` and `join` to make it slightly more easier constructing the filter.

```js
var ids = ["id1", "id2", "id3"];

await pb.collection("example").getList(1, 30, {
    filter: ids.map((id) => `id='${id}'`).join("||"),
})
```

**Reply by lbensaad** [reply]
Understood, many thanks

**Reply by Christophershivers** [reply]
is there going to be support for IN in the future?

**Reply by ganigeorgiev** [reply]
@Christophershivers It exists in the roadmap but remains a low priority.

**Comment by lbensaad** [comment]
I tried your suggestion, but the problem is that if the ids array is empty then the map result will be empty string and the filter will be empty and it will list all records.

**Reply by ganigeorgiev** [reply]
The behavior would be practically the same if there was a `IN` operator.

If you know upfront that there are no ids to lookup you can directly skip the request.
